### PR TITLE
Fix `NamedExpr` location

### PR DESCRIPTION
**Pull request by harupy**
Fixes `NamedExpr` location.

### CODE REVIEW

File: compiler/parser/python.lalrpop [hunk 1]
```diff
@@ -767,7 +767,7 @@ NamedExpressionTest: ast::Expr = {
 }
 
 NamedExpression: ast::Expr = {
-    <location:@L> <id:Identifier> ":=" <value:Test<"all">> <end_location:@R> => {
+    <location:@L> <id:Identifier> ":=" <end_location:@R> <value:Test<"all">> => {
```
Comment (harupy) [inline compiler/parser/python.lalrpop:770]:
```
>>> print(ast.dump(ast.parse("if (x := 1): pass"), include_attributes=True, indent=2))
Module(
  body=[
    If(
      test=NamedExpr(
        target=Name(
          id='x',
          ctx=Store(),
          lineno=1,
          col_offset=4,
          end_lineno=1,
          👇
          end_col_offset=5),
        value=Constant(
          value=1,
          lineno=1,
          col_offset=9,
          end_lineno=1,
          end_col_offset=10),
        lineno=1,
        col_offset=4,
        end_lineno=1,
        end_col_offset=10),
      body=[
        Pass(
          lineno=1,
          col_offset=13,
          end_lineno=1,
          end_col_offset=17)],
      orelse=[],
      lineno=1,
      col_offset=0,
      end_lineno=1,
      end_col_offset=17)],
  type_ignores=[])
```

Comment (harupy) [inline compiler/parser/python.lalrpop:770]:
```suggestion
    <location:@L> <id:Identifier> <end_location:@R> ":=" <value:Test<"all">> => {
```

**Comment by harupy** [comment]
Wait

**Review by youknowone — APPROVED** [review]
Thank you!

**Pull request merged**